bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxPowerlinkMessage.java

```diff
@@ -50,37 +48,29 @@ public PowermaxState handleMessage(PowermaxCommManager commManager) {
 
         if (subType == 0x03) {
             // keep alive message
+
+            addDebugInfo("Subtype", subType, "Keep Alive");
+
             commManager.sendAck(this, (byte) 0x02);
             updatedState = commManager.createNewState();
             updatedState.setLastKeepAlive(System.currentTimeMillis());
-        } else if (subType == 0x0A && message[4] == 0x01) {
-            logger.debug("Powermax alarm binding: Enrolling Powerlink");
-            commManager.enrollPowerlink();
-            updatedState = commManager.createNewState();
-            updatedState.setDownloadSetupRequired(true);
-        } else {
-            commManager.sendAck(this, (byte) 0x02);
-        }
-
-        return updatedState;
-    }
-
-    @Override
-    public String toString() {
-        String str = super.toString();
+        } else if (subType == 0x0A) {
+            byte enroll = message[4];
 
-        byte[] message = getRawData();
-        byte subType = message[2];
+            addDebugInfo("Subtype", subType, "Enroll");
+            addDebugInfo("Enroll", enroll);
```

Comment:
This should be moved in the next if

Reply:
Can `enroll` be anything other than 0x01? Does it send a 0x00 for an unenrollment for example? There's another message inside the conditional, so I kept this one out to always log the value of the `enroll` parameter.

Reply:
I don't know if enroll can be other than 1. But it is safer t consider it can be.